homeassistant/components/sentry/config_flow.py

```diff
@@ -13,34 +14,24 @@
 DATA_SCHEMA = vol.Schema({vol.Required(CONF_DSN): str})
 
 
-async def validate_input(hass: core.HomeAssistant, data):
-    """Validate the DSN input allows us to connect.
-
-    Data has the keys from DATA_SCHEMA with values provided by the user.
-    """
-    # validate the dsn
-    Dsn(data["dsn"])
-
-    return {"title": "Sentry"}
-
-
 class DomainConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a Sentry config flow."""
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
 
-    async def async_step_user(self, user_input=None):
+    async def async_step_user(
+        self, user_input: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, Any]:
         """Handle a user config flow."""
         if self._async_current_entries():
             return self.async_abort(reason="already_configured")
 
         errors = {}
         if user_input is not None:
             try:
-                info = await validate_input(self.hass, user_input)
-
-                return self.async_create_entry(title=info["title"], data=user_input)
+                Dsn(user_input["dsn"])
+                return self.async_create_entry(title="Sentry", data=user_input)
```

Comment:
Please move the entry creation to an `else:` block so we only wrap the code that can raise in the try... except.

Reply:
Done